kolibri/core/assets/src/views/tab-link/index.vue

```diff
@@ -0,0 +1,114 @@
+<template>
+
+  <li>
+    <router-link
+      :to="link"
+      class="tab"
+      :class="{ selected: selected, 'tab-icon-title': icon && title }">
+
+      <div
+        v-if="icon"
+        class="tab-icon"
+      >
+        <ui-icon
+          :icon="icon"
+          class="icon"
+        />
+      </div>
+
+      <div
+        v-if="title"
+        class="tab-title">
+        {{ title }}
+      </div>
+
+    </router-link>
+  </li>
+
+</template>
+
+
+<script>
+
+  module.exports = {
+    props: {
+      title: {
+        type: String,
+        required: false,
+      },
+      icon: {
+        type: String,
+        required: false,
+      },
+      link: {
+        type: [String, Object],
+        required: true,
+      },
```

Comment:
See an example of a validator used elsewhere: https://github.com/learningequality/kolibri/blob/63cca6edea924141c3f5da60ddef0ccbb2f7ecfd/kolibri/plugins/coach/assets/src/views/reports/table-cells/name-cell.vue#L36

Recommend following this pattern. If we want to accept strings also, we should update that validator accordingly so it's consistent across the application - but I don't see a need for it at this point. Better to be more restrictive with the API until a need is shown.